changelogs/fragments/8383-deprecate-gitlab-basic-auth.yml

```
@@ -0,0 +1,2 @@
+deprecated_features:
+  - "gitlab modules - the basic auth method on GitLab API have been deprecated and will be removed in community.general 9.0.0 (https://github.com/ansible-collections/community.general/pull/8383)."
```

The next feature release is 9.0.0, so the removal version needs to be at least one major version later.
```suggestion
  - "gitlab modules - the basic auth method on GitLab API have been deprecated and will be removed in community.general 10.0.0 (https://github.com/ansible-collections/community.general/pull/8383)."
```